src/tx/builder/field-types/deposit.ts

```diff
@@ -8,14 +8,16 @@ export default {
   /**
    * @param value - Deposit value in string format. Should be equal to '0'.
    * @param options - Options
+   * @param parameters - Parameters
    * @returns Deposit value Buffer.
    */
   serialize(
     value: Int | undefined,
     options: Parameters<typeof coinAmount['serialize']>[1],
+    parameters: Parameters<typeof coinAmount['serialize']>[2],
```

Comment:
Maybe we could have a type alias with a readable name for such type builds. `Parameters<typeof coinAmount['serialize']>[2]`

Reply:
This is quite verbose, but the base idea is just to copy the signature of the inherited method. I'm doing it frequently with different objects and their parts, so I'm not sure that adding intermediate types would be helpful in all cases, but I'm doing so in the most popular, like this:
https://github.com/aeternity/aepp-sdk-js/blob/6b8888b7d8bb168f2f3dd0f9737e88dce4889ffe/src/tx/index.ts#L34-L35